src/platform/debugger/jupyter/kernelDebugAdapter.ts

```diff
@@ -275,6 +264,35 @@ export class KernelDebugAdapter implements DebugAdapter, IKernelDebugAdapter, ID
         return undefined;
     }
 
+    // Use our jupyter session to delete all the cells
+    private async deleteDumpCells() {
+        const fileValues = [...this.cellToFile.values()];
+        // Need to have our Jupyter Session and some dumpCell files to delete
+        if (this.jupyterSession && fileValues.length) {
+            // Create our python string of file names
+            const fileListString = fileValues
+                .map((filePath) => {
+                    return '"' + filePath + '"';
+                })
+                .join(',');
+
+            // Insert into our delete snippet
+            const deleteFilesCode = `import os
+_VSCODE_fileList = [${fileListString}]
+for file in _VSCODE_fileList:
+    try:
+        os.remove(file)
+    except:
+        pass
+del _VSCODE_fileList`;
+
+            return executeSilently(this.jupyterSession, deleteFilesCode, {
```

Comment:
Any idea why the execution is set to `silent: false`, is this intentional https://github.com/microsoft/vscode-jupyter/blob/bf8e4adebaa4da1fd7923eed536e9ce0ffe6f788/src/kernels/helpers.ts#L1452 ?

Reply:
Yeah, as I recall the name is bad. Silent also swallows all output and we want to see if there are errors coming in on stdout. It's actually store_history that makes it silent in terms of hiding the execution from execution count.

Reply:
Jupyter has doc on the execute params here which detail it a bit more: https://jupyter-client.readthedocs.io/en/latest/messaging.html#execute

Reply:
@IanMatthewHuff thanks for the detailed explanation, you are right `store_history` is the key here and our `executeSilently` here means skipping history and execution count.